app/models/rule.rb

```diff
@@ -33,10 +33,20 @@ class Rule < ApplicationRecord
   has_many :hosts, through: :rule_results, source: :host
   has_many :rule_references_rules, dependent: :delete_all
   has_many :rule_references, through: :rule_references_rules
+  has_many :required_for_rule_group_rules, class_name: 'RuleGroupRule',
```

Comment:
Why the `_for_` is it adding anything semantically?

Reply:
I named this association `required_for_rule_group_rules` because if you call this association on a rule group/rule, like `test_rule_group.required_for_rule_group_rules`, it wouldn't return the required rules/rule_groups for `test_rule_group` but it returns the rules and rule_groups that require `test_rule_group`

Reply:
Hm, I'm wondering if this even necessary for us :thinking:

Reply:
Oh yeah, that's true.. Now that I think of it, we'll probably only really use the `included_rule_group_rules` because we can get the conflicted/required rules or rule groups from there